phoenix-core/src/it/java/org/apache/phoenix/end2end/ExplainPlanWithMultipleRowsPerGuidePostIT.java

```diff
@@ -0,0 +1,303 @@
+        // (UNBOUND, 139]   (139 179]   (179, 219]   (219, 259]   (259, 299]   (299, 339]   (339, 379]   (379, 419]   (419, 459]   (459, 499]   (499, UNBOUND)
+        List<Object> binds = Lists.newArrayList();
+        try (Connection conn = DriverManager.getConnection(getUrl())) {
+            int[] result;
+            Long estimatedBytesPerRow = 100L;
+
+            // Query without any filter. Query Key Range (UNBOUND, UNBOUND).
+            String sql = "SELECT a FROM " + tableName;
+            ResultSet rs = conn.createStatement().executeQuery(sql);
+            int i = 0;
+            int numRows = 400;
+            int estimatedRows = 400;
+            while (rs.next()) {
+                assertEquals(100 + i, rs.getInt(1));
+                i++;
+            }
+            assertEquals(numRows, i);
+            // Hit all guide posts.
+            Estimate info = getByteRowEstimates(conn, sql, binds);
+            assertEquals(Long.valueOf(estimatedRows), info.getEstimatedRows());
+            assertEquals((Long) (estimatedRows * estimatedBytesPerRow), info.getEstimatedBytes());
+            assertTrue(info.getEstimateInfoTs() > 0);
+
+
+            // Query whose start key is before any data. Query Key Range [99, UNBOUND).
+            sql = "SELECT a FROM " + tableName + " WHERE K >= 99";
+            rs = conn.createStatement().executeQuery(sql);
+            i = 0;
+            numRows = 400;
+            estimatedRows = 400;
+            while (rs.next()) {
+                assertEquals(100 + i, rs.getInt(1));
+                i++;
+            }
+            assertEquals(numRows, i);
+            // Hit all guide posts.
+            info = getByteRowEstimates(conn, sql, binds);
+            assertEquals(Long.valueOf(estimatedRows), info.getEstimatedRows());
+            assertEquals((Long) (estimatedRows * estimatedBytesPerRow), info.getEstimatedBytes());
+            assertTrue(info.getEstimateInfoTs() > 0);
+
+            // Query whose start key is after any data. Query Key Range [500, UNBOUND).
+            sql = "SELECT a FROM " + tableName + " WHERE K >= 500";
+            rs = conn.createStatement().executeQuery(sql);
+            assertFalse(rs.next());
+            // Hit zero guide posts.
+            info = getByteRowEstimates(conn, sql, binds);
+            estimatedRows = 0;
+            assertEquals(Long.valueOf(estimatedRows), info.getEstimatedRows());
+            assertEquals((Long) (estimatedRows * estimatedBytesPerRow), info.getEstimatedBytes());
+            assertTrue(info.getEstimateInfoTs() > 0);
+
+            // Query whose end key is before any data. Query Key Range (UNBOUND, 98].
+            sql = "SELECT a FROM " + tableName + " WHERE K <= 98";
+            rs = conn.createStatement().executeQuery(sql);
+            assertFalse(rs.next());
+            // Hit the first guide post.
+            info = getByteRowEstimates(conn, sql, binds);
+            estimatedRows = 10;
```

Comment:
We didn't insert rows before 100, why we are getting 10 here?

Reply:
Because the query will hit the first guide post which has key 100. This guide post contains 10 estimated rows.